addon/background.js

```diff
@@ -0,0 +1,206 @@
+      if (k === "environment") {
+        currentEnvironment = changes[k];
+        fetchResources();
+      }
+    });
+  });
+}
+
+function fetchResources() {
+  log("fetchResources");
+  return Promise.all(
+    Object.keys(resources).map(path =>
+                               fetch(`${currentEnvironment.baseUrl}/api/${path}.json`)
+                               .then(res => res.json())
+                               .then((data) => data.results ? data.results : data)
+                               .then((data) => [path, data])
+                               .catch(err => {
+                                 log("fetchResources error", path, err);
+                                 return [path, null];
+                               })
+                              )
+  ).then(results => {
+    log("fetchResources results", results);
+    results.forEach(([path, data]) => (resources[path] = data));
+    updateBadgeTextOnNew(resources.experiments, resources.news_updates);
+    log("fetchResources results parsed", resources);
+  });
+}
+
+async function updateBadgeTextOnNew(experiments, news_updates) {
+  let { clicked } = await storage.get("clicked");
+  if (!clicked) {
+    // Set initial button click timestamp if not found
+    clicked = Date.now();
+    await storage.set({ clicked });
+  }
+
+  const newExperiments = (experiments || []).filter(experiment => {
+    const dt = new Date(experiment.modified || experiment.created).getTime();
+    return dt >= clicked;
+  });
+
+  // check for port number on local, we need to strip it off
+  // to properly fetch cookies.
+  const baseUrl = currentEnvironment.baseUrl;
+  const portIndex = baseUrl.indexOf(":8000");
+  const cookieUrl = (portIndex > -1) ? baseUrl.substring(0, portIndex) : baseUrl;
+
+  let lastViewed = 0;
+  const cookie = await browser.cookies.get({
+    url: cookieUrl,
+    name: "updates-last-viewed-date"
+  });
+
+  if (cookie) lastViewed = cookie.value;
+
+  /* only show badge for news update if:
+   * - has the "major" key
+   * - update has been published in the past two weeks
+   * - update has not been "seen" by the frontend (lastViewed)
+   * - update has not been "seen" by the addon (clicked)
+   */
+  const twoWeeksAgo = Date.now() - TWO_WEEKS;
+  const newsUpdates = (news_updates || []).filter((u) => u.major)
+        .filter((u) => new Date(u.published).getTime() >= twoWeeksAgo)
+        .filter((u) => new Date(u.published).getTime() >= new Date(lastViewed))
+        .filter((u) => new Date(u.published).getTime() >= clicked);
+
+  BROWSER_ACTION_LINK = (newExperiments.length || newsUpdates.length) > 0
+    ? BROWSER_ACTION_LINK_BADGED
+    : BROWSER_ACTION_LINK_NOT_BADGED;
+
+  browser.browserAction.setBadgeText({
+    text: (newExperiments.length || newsUpdates.length) > 0 ? "!" : ""
+  });
+
+  browser.browserAction.setBadgeText({
+    text: (newExperiments.length || newsUpdates.length) > 0 ? "!" : ""
+  });
+}
+
+function submitPing(object, event, addons) {

```

Comment:
Telemetry and ping centre are no longer called in the new webextension. We're only using google analytics now.